fvm/transactionInvoker.go

```diff
@@ -75,8 +75,27 @@ func (i *TransactionInvoker) Process(
 			proc.Events = make([]flow.Event, 0)
 			proc.ServiceEvents = make([]flow.Event, 0)
 		}
+
 		if mergeError := parentState.MergeState(childState, sth.EnforceInteractionLimits()); mergeError != nil {
-			processErr = fmt.Errorf("transaction invocation failed when merging state: %w", mergeError)
+			if processErr != nil {
```

Comment:
This seems unnecessary.  Why not just report format both the merge error string and the original process error string into the new processErr?